### PR TITLE
PDS-Test: DeleteDataServiceAndValidateBackupAtObjectStore 

### PR DESCRIPTION
**Pull request by gauraa2**
Summary: This test validates the deletion the PDS data service should not delete the backups in backend.
Logs: https://aetos.pwx.purestorage.com/resultSet/testSetID/225115
Full logs: https://gist.github.com/gauraa2/f3786a6ec59a980040b0795a26e9b95c

### CODE REVIEW

File: tests/pds/backup_test.go [hunk 1]
```diff
@@ -72,3 +142,13 @@ func deleteAllBkpTargets() {
 		log.FailOnError(err, "Failed while deleting the GCP backup target")
 	}
 }
+
+func randString(length int) string {
```
Comment (dbhatnagar-px) [inline tests/pds/backup_test.go:146]: This is a utility function so should go in utils

**Review by dbhatnagar-px — CHANGES_REQUESTED** [review]

File: tests/pds/backup_test.go (hunk 1 above)
Comment (gauraa2) [reply tests/pds/backup_test.go]: Done

File: tests/pds/common.go [hunk 2]
```diff
@@ -73,6 +73,7 @@ var (
 	dataServiceDefaultResourceTemplateIDMap map[string]string
 	dataServiceNameIDMap                    map[string]string
 	supportedDataServicesNameIDMap          map[string]string
+	backupSupportedDataServiceNameIDMap     map[string]string
 	DeployAllVersions                       bool
 	DataService                             string
 	DeployAllImages                         bool
```
[the remark this answers is not in the document]
Comment (gauraa2) [reply tests/pds/common.go:80]: Done

File: tests/pds/backup_test.go [hunk 3]
```diff
@@ -57,6 +58,75 @@ var _ = Describe("{ValidateBackupTargetsOnSupportedObjectStores}", func() {
...
+					}
+					for _, folder := range folders {
+						log.Infof("Found the folder: %v (default encrypted).", folder)
+					}
+				})
+			}
+		})
+	})
+	JustAfterEach(func() {
+		defer EndTorpedoTest()
```
[the remark this answers is not in the document]
Comment (gauraa2) [reply tests/pds/backup_test.go]: Done

**Review by dbhatnagar-px — APPROVED** [review]
looks good

File: tests/pds/backup_test.go [hunk 4]
```diff
@@ -57,6 +58,77 @@ var _ = Describe("{ValidateBackupTargetsOnSupportedObjectStores}", func() {
...
+					log.InfoD("Data service: %v doesn't support backup, skipping...", ds.Name)
+					continue
+				}
+				currentTimestamp := time.Now().UTC()
+				stepLog = "Deploy and validate data service"
+				Step(stepLog, func() {
+					log.InfoD(stepLog)
+					deployment, _, _, err = DeployandValidateDataServices(ds, params.InfraToTest.Namespace, tenantID, projectID)
+					log.FailOnError(err, "Error while deploying data services")
+				})
```
Comment (Madanagopal19) [inline tests/pds/backup_test.go:95]: Add a TODO for adding a defer block to clean up deployments

File: drivers/pds/pdsbackup/backuputils.go [hunk 5]
```diff
@@ -237,16 +306,16 @@ func InitializePdsBackup() (*BackupClient, error) {
 	apiConf.Scheme = endpointURL.Scheme
 
 	apiClient := pds.NewAPIClient(apiConf)
-	components := pdsapi.NewComponents(apiClient)
```
Comment (Madanagopal19) [inline drivers/pds/pdsbackup/backuputils.go:240]: Add a TODO to use a Init method in pds.go

File: drivers/pds/pdsbackup/backuputils.go [hunk 6]
```diff
@@ -3,6 +3,7 @@ package pdsbackup
 import (
```
Comment (Madanagopal19) [inline drivers/pds/pdsbackup/backuputils.go:3]: nit: rename the file to backup_utils.go

**Review by Madanagopal19 — CHANGES_REQUESTED** [review]
minor comments

File: drivers/pds/pdsbackup/backuputils.go (hunk 5 above)
Comment (gauraa2) [reply drivers/pds/pdsbackup/backuputils.go:240]: Done

File: drivers/pds/pdsbackup/backuputils.go (hunk 6 above)
Comment (gauraa2) [reply drivers/pds/pdsbackup/backuputils.go:3]: Done

**Review by Madanagopal19 — APPROVED** [review]
LGTM

**Review by aravindranath14 — APPROVED** [review]
LGTM

**Pull request merged**